wasm-runtime/polkadot/src/support/storable.rs

```diff
@@ -17,29 +17,48 @@
 //! Stuff to do with the runtime's storage.
 
 use slicable::Slicable;
-use endiansensitive::EndianSensitive;
 use keyedvec::KeyedVec;
 use runtime_support::prelude::*;
 use runtime_support::{self, twox_128};
 
 /// Trait for a value which may be stored in the storage DB.
 pub trait Storable {
 	/// Lookup the value in storage and deserialise, giving a default value if not found.
-	fn lookup_default(key: &[u8]) -> Self where Self: Sized + Default { Self::lookup(key).unwrap_or_else(Default::default) }
+	fn lookup_default(key: &[u8]) -> Self where Self: Sized + Default {
+		Self::lookup(key).unwrap_or_else(Default::default)
+	}
+
 	/// Lookup `Some` value in storage and deserialise; `None` if it's not there.
-	fn lookup(_key: &[u8]) -> Option<Self> where Self: Sized { unimplemented!() }
+	fn lookup(_key: &[u8]) -> Option<Self> where Self: Sized {
+		unimplemented!()
+	}
+
+	/// Retrives and returns the serialised value of a key from storage, removing it immediately.
+	fn take(key: &[u8]) -> Option<Self> where Self: Sized {
+		if let Some(value) = Self::lookup(key) {
```

Comment:
Using `Self` in this way is pretty unidiomatic. A trait should provide the core functionality and then a module can expose free functions to leverage it.

Reply:
storage API refactor can happen in later PR.